src/linechart/DataSeries.jsx

```diff
@@ -2,139 +2,100 @@
 
 var React = require('react');
 var d3 = require('d3');
+var VoronoiCircleContainer = require('./VoronoiCircleContainer');
 var Line = require('./Line');
-var Circle = require('./Circle');
-
 
 module.exports = React.createClass({
 
   displayName: 'DataSeries',
 
   propTypes: {
+    color: React.PropTypes.func,
+    colorAccessor: React.PropTypes.func,
     data: React.PropTypes.array,
-    seriesName: React.PropTypes.string,
     interpolationType: React.PropTypes.string,
-    fill: React.PropTypes.string,
     xAccessor: React.PropTypes.func,
     yAccessor: React.PropTypes.func,
-    displayDataPoints: React.PropTypes.bool
   },
 
   getDefaultProps() {
     return {
       data: [],
-      seriesName: "",
-      interpolationType: 'linear',
-      fill: '#fff',
       xAccessor: (d) => d.x,
       yAccessor: (d) => d.y,
-      displayDataPoints: true
+      interpolationType: 'linear'
     };
   },
-
+  
   _isDate(d, accessor) {
-    return Object.prototype.toString.call(accessor(d)) === '[object Date]';
+      return Object.prototype.toString.call(accessor(d)) === '[object Date]';
   },
 
   render() {
-
     var props = this.props;
-
-    // Check to see if there is any data to draw lines/circles with
-    // Return null if there is nothing to draw (null will become <noscript>)
-    if (props.data.length === 0) {
```

Comment:
This check is still needed since below we use the first element in the array to figure out if we need to use dates or not.